app/services/foreman_ansible/inventory_creator.rb

```diff
@@ -38,10 +38,18 @@ def connection_params(host)
       params = {
         'ansible_port' => host_port(host),
         'ansible_user' => host_user(host),
-        'ansible_ssh_pass' => host_ssh_pass(host),
+        'ansible_become' => host_become(host),
         'ansible_connection' => connection_type(host),
         'ansible_winrm_server_cert_validation' => winrm_cert_validation(host)
       }
+      # Check if private_key is defined, if it is use that. Otherwise
+      # use ssh_pass
+      if !host.host_params['ansible_ssh_private_key_file'].empty? ||
+          Setting[:ansible_ssh_private_key_file].empty?
```

Comment:
I think the `!` got lost, it was here before. You can keep `if`, I do not mind either way. My main point was that you can re-use the logic from `host_private_key_file` method and be more DRY:

```ruby
if !host_private_key_file(host).empty?
```